forge/test/mlir/test_ops.py

```diff
@@ -558,3 +558,32 @@ def forward(self, x):
     
     co_out = [co.to("cpu") for co in co_out]
     assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
+
+@pytest.mark.parametrize("dim", [-1, -2, -3], ids=["-1", "-2", "-3"])
+@pytest.mark.parametrize("start", [0],  ids=["0"])
+@pytest.mark.parametrize("stop", [2,32,64], ids=["2", "32", "64"])
+@pytest.mark.parametrize("stride", [1, 2, 4, 8], ids=["1", "2", "4", "8"])
+@pytest.mark.parametrize("shape", [(1, 32, 64, 64), (32, 64, 64), (64, 64)])
+def test_indexing(dim, start, stop, stride, shape):
+    if len(shape) == 2 and dim == -3:
+        pytest.skip("Skipping since indexing on dim=-3, 2D tensor doesn't make sense")
+    if stop > shape[dim]:
+        pytest.skip("Skipping since stop > shape[dim]")
+    class ForgeIndexing(forge.ForgeModule):
+        def __init__(self, dim, start, stop, stride):
+            super().__init__("ForgeTest")
+
+        def forward(self, x):
+            return forge.op.Index("indexing_op_1", x, dim, start, stop, stride)
```

Comment:
Can we use PyTorch op instead of native Index op on our end?

Reply:
Did you try to run pure indexing over it? E.g. `input_tensor[:,:,1]`? To what is it lowered through TVM?

Reply:
I tried now, and I get some error in generate graph: `AttributeError: 'Tensor' object has no attribute 'src_op'`. I can take a look into this but it seems like a problem with generating initial graph...

Reply:
Both pure indexing and index_select yield the same error. I have to look into this, but I wouldn't delay this mr because of that. What do you think?

Reply:
I agree. Feel free to create a new issue to tackle with this problem. I don't think this one is a blocker for Llama 3B.